src/main/kotlin/org/rust/ide/ui/layout.kt

```diff
@@ -54,6 +55,17 @@ private class RsLayoutBuilderImpl(
         }
         labeledComponent.toolTipText = toolTip.trimIndent()
         labeledComponents += labeledComponent
-        panel.add(labeledComponent)
+
+        if (component.isVisible) {
+            panel.add(labeledComponent)
+        }
+    }
+
+    override fun component(component: JComponent) {
+        val innerPanel = JPanel(BorderLayout()).apply {
+            border = JBUI.Borders.empty(VERTICAL_OFFSET, HORIZONTAL_OFFSET)
```

Comment:
nit: if you extract code from `apply`, it will take 2 lines instead of 3 and in my opinion will be a little bit more readable